### PR TITLE
fix: pdf tabs

### PR DESCRIPTION
**Pull request by simboonlong**
### pr update
- only render tab if attachment type is pdf

### user story
- https://www.pivotaltracker.com/story/show/173483758

### preview 
Screenshot 2020-06-26 at 3 34 52 PM

### CODE REVIEW

File: src/components/CertificateViewer.tsx [hunk 1]
```diff
@@ -34,16 +34,31 @@ export const CertificateViewer = ({
...
-    setTemplates(templates);
-    setSelectedTemplate(templates[0].id);
-  }, []);
+  const updateTemplates = useCallback(
+    (templates) => {
+      setTemplates(templates);
+      setSelectedTemplate(templates[0].id);
+
+      // modify tabs by reseting templates if attachment(s) exists
+      if (hasAttachments) {
```
Comment (Nebulis) [inline src/components/CertificateViewer.tsx]: Why not just running that branch always? I don't understand the need to check if there are attachments it did the same with a filter, just always run the filter

File: src/components/DecentralisedTemplateRenderer/DecentralisedRenderer.tsx [hunk 2]
```diff
@@ -7,7 +7,7 @@ import { LEGACY_OPENCERTS_RENDERER } from "../../config";
 
 interface DecentralisedRendererProps {
   rawDocument: WrappedDocument;
-  updateTemplates: (templates: { id: string; label: string }[]) => void;
+  updateTemplates: (templates: { id: string; label: string; type: string }[]) => void;
```
Comment (Nebulis) [inline src/components/DecentralisedTemplateRenderer/DecentralisedRenderer.tsx]: Maybe it's time to create a type... In the lib? Can start in the app first and move later (one day) in the lib

File: src/test/attachments.spec.js [hunk 3]
```diff
@@ -7,19 +7,31 @@ const Document = "./fixture/attachments.json";
 const TabsItems = Selector("[data-testid='tabs'] .nav-item");
```
Comment (Nebulis) [inline src/test/attachments.spec.js:7]: Should we add a test where a file contain attachments that should be hidden?

Comment (simboonlong) [reply src/test/attachments.spec.js:7]: ok sure

File: src/components/CertificateViewer.tsx [hunk 4]
```diff
@@ -34,15 +35,21 @@ export const CertificateViewer = ({
...
   const originalData = getData(document);
   const attachments = originalData?.attachments;
   const hasAttachments = attachments && attachments.length > 0;
 
   const updateTemplates = useCallback((templates) => {
-    setTemplates(templates);
-    setSelectedTemplate(templates[0].id);
+    // filter all templates that are renderable currently
+    const templatesModified = templates.filter((item: { type: string }) => {
+      return item.type === "custom-template" || item.type === "application/pdf";
```
Comment (Nebulis) [inline src/components/CertificateViewer.tsx]: just to be sure, that means that renderer before 2.3.0 will not display anything (no type)

**Review by Nebulis — APPROVED** [review]

File: src/components/CertificateViewer.tsx (hunk 4 above)
Comment (simboonlong) [reply src/components/CertificateViewer.tsx]: as discussed, will cater to renderer before 2.3.0. made the necessary edits

**Review by Nebulis — APPROVED** [review]

**Pull request merged**